components/membership/registration-impl/src/test/kotlin/net/corda/membership/impl/registration/dynamic/mgm/MGMRegistrationServiceTest.kt

```diff
@@ -0,0 +1,208 @@
+    private val registrationService = MGMRegistrationService(
+        publisherFactory,
+        configurationReadService,
+        lifecycleCoordinatorFactory,
+        layeredPropertyMapFactory,
+        cryptoOpsClient,
+        keyEncodingService
+    )
+
+    private val properties = mapOf(
+        "corda.session.key.id" to SESSION_KEY_ID,
+        "corda.ecdh.key.id" to ECDH_KEY_ID,
+        "corda.group.protocol.registration" to "net.corda.membership.impl.registration.dynamic.MemberRegistrationService",
+        "corda.group.protocol.synchronisation" to "net.corda.membership.impl.sync.dynamic.MemberSyncService",
+        "corda.group.protocol.p2p.mode" to "AUTHENTICATED_ENCRYPTION",
+        "corda.group.key.session.policy" to "Combined",
+        "corda.group.pki.session" to "Standard",
+        "corda.group.pki.tls" to "C5",
+        "corda.endpoints.0.connectionURL" to "localhost:1080",
+        "corda.endpoints.0.protocolVersion" to "1",
+        "corda.group.truststore.session.0" to "-----BEGIN CERTIFICATE-----Base64–encoded certificate-----END CERTIFICATE-----",
+        "corda.group.truststore.tls.0" to "-----BEGIN CERTIFICATE-----Base64–encoded certificate-----END CERTIFICATE-----",
+    )
+
+    @Suppress("UNCHECKED_CAST")
+    private fun setUpPublisher() {
+        // kicks off the MessagingConfigurationReceived event to be able to mock the Publisher
+        registrationServiceLifecycleHandler?.processEvent(
+            ConfigChangedEvent(setOf(ConfigKeys.BOOT_CONFIG, ConfigKeys.MESSAGING_CONFIG), TestUtils.configs),
+            coordinator
+        )
+    }
+
+    @Test
+    fun `starting and stopping the service succeeds`() {
+        registrationService.start()
+        assertTrue(registrationService.isRunning)
+        registrationService.stop()
+        assertFalse(registrationService.isRunning)
+    }
+
+    @Test
+    fun `registration successfully builds MGM info and publishes it`() {
+        setUpPublisher()
+        registrationService.start()
+        val capturedPublishedList = argumentCaptor<List<Record<String, Any>>>()
+        val result = registrationService.register(mgm, properties)
+        verify(mockPublisher, times(1)).publish(capturedPublishedList.capture())
+        val publishedMgmInfoList = capturedPublishedList.firstValue
+        assertSoftly {
+            it.assertThat(result.outcome).isEqualTo(MembershipRequestRegistrationOutcome.SUBMITTED)
+            it.assertThat(publishedMgmInfoList.size).isEqualTo(1)
+            val publishedMgmInfo = publishedMgmInfoList.first()
+            it.assertThat(publishedMgmInfo.topic).isEqualTo(Schemas.Membership.MEMBER_LIST_TOPIC)
+            val expectedRecordKey = "${mgmId}-${mgmId}"
+            it.assertThat(publishedMgmInfo.key).isEqualTo(expectedRecordKey)
+            val persistentMemberPublished = publishedMgmInfo.value as PersistentMemberInfo
+            val mgmPublished = memberInfoFactory.create(
```

Comment:
`memberInfoFactory` has a create function for `PersistentMemberInfo`
```suggestion
            val mgmPublished = memberInfoFactory.create(persistentMemberPublished)
```